src/coreclr/src/jit/liveness.cpp

```diff
@@ -2411,37 +2411,23 @@ void Compiler::fgInterBlockLocalVarLiveness()
 
     for (block = fgFirstBB; block; block = block->bbNext)
     {
-        if (block->bbCatchTyp != BBCT_NONE)
+        if (block->hasEHBoundaryIn())
         {
             /* Note the set of variables live on entry to exception handler */
             VarSetOps::UnionD(this, exceptVars, block->bbLiveIn);
         }
 
-        if (block->bbJumpKind == BBJ_EHFILTERRET)
-        {
-            /* Get the set of live variables on exit from a 'filter' */
-            VarSetOps::UnionD(this, filterVars, block->bbLiveOut);
-        }
-        else if (block->bbJumpKind == BBJ_EHFINALLYRET)
-        {
-            /* Get the set of live variables on exit from a 'finally' block */
-
-            VarSetOps::UnionD(this, finallyVars, block->bbLiveOut);
-        }
-#if defined(FEATURE_EH_FUNCLETS)
-        // Funclets are called and returned from, as such we can only count on the frame
-        // pointer being restored, and thus everything live in or live out must be on the
-        // stack
-        if (block->bbFlags & BBF_FUNCLET_BEG)
-        {
-            VarSetOps::UnionD(this, exceptVars, block->bbLiveIn);
```

Comment:
You removed this, but the `hasEHBoundaryIn()` case above doesn't replace it for filter/finally/fault funclets.

Reply:
I believe that you had indicated to me that a block with `BBF_FUNCLET_BEG` set would always have `bbCatchTyp != BBCT_NONE`, so this was changed to an assert for the `hasEHBoundaryIn()` case - is there something else I've missed?

Reply:
Ah, right, I forgot about the special BBCT values for fault/finally/filter/filter-handler.